services/mailservice/mail.go

```diff
@@ -25,6 +25,14 @@ import (
 	"github.com/mattermost/mattermost-server/utils"
 )
 
+// SMTPClient is implemented by an smtp.Client. See https://golang.org/pkg/net/smtp/#Client.
+//
+type SMTPClient interface {
```

Comment:
Excellent pattern! Would it make sense to leave this interface unexported? (The client need merely implement it, but never cares about using `SMTPClient` itself.)

Reply:
Good call.